api/auth.mjs

```diff
@@ -0,0 +1,154 @@
+import { BAD_REQUEST, UNAUTHORIZED, TYPE_JSON, INTERNAL, rand } from './defs.mjs'
+import { GET, withUser } from './router.mjs'
+import * as db from './db.mjs'
+
+const gql = async (query) => {
+  const text = await query.text()
+  try {
+    const { data, errors } = JSON.parse(text)
+    return { data, error: errors && errors[0].message }
+  } catch {
+    return { error: text }
+  }
+}
+
+const GUILD = '821033851153285170'
+const DISCORD = 'https://discordapp.com/api'
+GET.auth.discord = async ({ url }) => {
+  const code = url.searchParams.get('code')
+  const state = url.searchParams.get('state')
+  if (!code || !state) return new Response('Missing Params', BAD_REQUEST)
+  const session = await db.get(`discord:${state}`)
+  if (!session?.user) return new Response('Bad State', BAD_REQUEST)
+  const authResponse = await fetch(`${DISCORD}/oauth2/token`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+    body: new URLSearchParams({
+      scope: 'identify gdm.join guilds.join',
+      client_id: '826974634069983282',
+      client_secret: DISCORD_SECRET,
+      grant_type: 'authorization_code',
+      // redirect_uri: redirectUri, do i need dis ?
+      code,
+    }),
```

Comment:
here I post the request to get the `OAuth2` token

Reply:
where are you find the client_id, and where is DISCORD_SECRET